### PR TITLE
AEA-826 Tac control contract skill fixes (part 1)

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Current tac_control_contract skill does not work. This PR reduces code duplication between the two tac control skills and makes the tac_control_contract skill almost working.

It also introduces miscellaneous fixes required to make it work.

It also adds some more linters.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/configurations/base.py [hunk 1]
```diff
@@ -1166,7 +1166,7 @@ class SkillConfig(ComponentConfiguration):
 
     default_configuration_filename = DEFAULT_SKILL_CONFIG_FILE
     package_type = PackageType.SKILL
-    configurable_fields = {"handlers", "behaviours", "models"}
+    configurable_fields = {"handlers", "behaviours", "models", "is_abstract"}
```
Comment (DavidMinarsch) [inline aea/configurations/base.py:1169]: @marcofavorito adding this here is not enough; how can this be extended so it is applied to the nested config?

File: docs/tac-skills.md [hunk 2]
```diff
@@ -206,8 +206,9 @@ aea add-key fetchai fetchai_private_key.txt --connection
 Navigate to the tac controller project, then use the command line to get and set the start time (set it to at least two minutes in the future):
 
 ``` bash
-aea config get vendor.fetchai.skills.tac_control.models.parameters.args.start_time
-aea config set vendor.fetchai.skills.tac_control.models.parameters.args.start_time '01 01 2020  00:01'
+aea config get vendor.fetchai.skills.tac_control.models.parameters.args.registration_start_time
+aea config set vendor.fetchai.skills.tac_control.models.parameters.args.registration_start_time '01 01 2020  00:01'
+aea config set vendor.fetchai.skills.tac_control.is_abstract false --type bool
```
Comment (DavidMinarsch) [inline docs/tac-skills.md:211]: revert

File: packages/fetchai/agents/tac_controller_contract/aea-config.yaml [hunk 3]
```diff
@@ -8,22 +8,51 @@ aea_version: '>=0.6.0, <0.7.0'
...
+  soef_addr: soef.fetch.ai
+  soef_port: 9002
+...
+---
+author: fetchai
+name: tac_control
+version: 0.7.0
+type: skill
+is_abstract: true
+...
```
Comment (DavidMinarsch) [inline packages/fetchai/agents/tac_controller_contract/aea-config.yaml:57]: @marcofavorito this is not currently supported by hashing script.

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**